packages/matchbox/src/components/Pagination/Pagination.js

```diff
@@ -1,119 +1,71 @@
-     */
-    onChange: PropTypes.func,
-
-    /**
-     * Hides first and last buttons
-     */
-    marginsHidden: PropTypes.bool,
-
-    /**
-     * Flat style buttons
-     */
-    flat: PropTypes.bool,
-
-    /**
-     * Selected page color
-     */
-    selectedColor: PropTypes.oneOf(['orange', 'blue', 'navy', 'purple', 'red']),
-  };
-
-  static defaultProps = {
-    currentPage: 1,
-    selectedColor: 'orange',
-  };
-
-  state = {
-    index: 0,
-    hasPrevious: false,
-    hasNext: true,
-  };
 
-  componentDidMount() {
-    this.handlePageChange(this.props.currentPage - 1);
+const StyledMoreHoriz = styled(MoreHoriz)`
+  fill: ${tokens.color_gray_500};
+`;
+
+function Pagination(props) {
+  const [index, setIndex] = React.useState(0);
+  const [hasPrevious, setHasPrevious] = React.useState(false);
+  const [hasNext, setHasNext] = React.useState(false);
+
+  const {
+    currentPage,
+    pages,
+    pageRange,
+    onChange,
+    marginsHidden,
+    flat,
+    selectedColor,
+    ...rest
+  } = props;
+
+  function handlePageChange(index) {
+    setIndex(index);
+    setHasPrevious(index > 0);
+    setHasNext(index < pages - 1);
+
+    onChange && onChange(index);
   }
 
-  componentDidUpdate(prevProps) {
-    const { pages, pageRange, currentPage } = this.props;
-    if (
-      pages !== prevProps.pages ||
-      pageRange !== prevProps.pageRange ||
-      currentPage !== prevProps.currentPage
-    ) {
-      this.handlePageChange(currentPage - 1);
-    }
+  function handleNext() {
+    handlePageChange(index + 1);
   }
 
-  handlePageChange(index) {
-    this.setState({
-      index,
-      hasPrevious: index > 0,
-      hasNext: index < this.props.pages - 1,
-    });
-
-    this.props.onChange && this.props.onChange(index);
+  function handlePrevious() {
+    handlePageChange(index - 1);
   }
 
-  handleNext = () => {
-    this.handlePageChange(this.state.index + 1);
-  };
-
-  handlePrevious = () => {
-    this.handlePageChange(this.state.index - 1);
-  };
-
-  _createButtons(start, end) {
+  function _createButtons(start, end) {
     const buttons = [];
 
-    for (let index = start; index < end; index++) {
+    for (let currentIndex = start; currentIndex < end; currentIndex++) {
       buttons.push({
-        content: `${index + 1}`,
-        onClick: this.handlePageChange.bind(this, index),
-        flat: this.props.flat,
-        className: classnames(
-          styles.Page,
-          index === this.state.index && styles.Selected,
-          index === this.state.index && styles[`color-${this.props.selectedColor}`],
-        ),
+        content: `${currentIndex + 1}`,
+        onClick: handlePageChange.bind(this, currentIndex),
+        flat: currentIndex === index ? false : true,
+        size: 'small',
+        color: currentIndex === index ? 'blue' : 'gray',
+        mx: '100',
+        width: currentIndex + 1 < 100 ? '32px' : 'auto',
       });
     }
 
     return buttonsFrom(buttons);
```

Comment:
Curious if you think we should keep using this function – I don't really have too much of an opinion but some prefer rendering JSX